### PR TITLE
Add wrapping logging class

### PR DESCRIPTION
**Pull request by rudigiesler**
Add a class that allows us to wrap existing log functions with ones that will add various arguments specified when we create the class.

### CODE REVIEW

**Comment by rudigiesler** [comment]
Ready for review.

File: vumi/tests/test_log.py [hunk 1]
```diff
@@ -45,3 +46,19 @@ def test_error_log_levels(self):
...
+
+        for method in (
+                log.debug, log.info, log.warning, log.error, log.critical,
+                log.msg, log.err):
+            with LogCatcher() as lc:
+                method('Test log')
+            [log] = lc.logs
+
+            self.assertEqual(log['system'], 'test')
+            self.assertEqual(log['bar'], 'foo')
```
Comment (hodgestar) [inline vumi/tests/test_log.py]: Should we perhaps also check that each method logs at the correct level? E.g. that `WrappingLogger.debug` actually sends a debug log message?

**Comment by hodgestar** [comment]
Left one comment, otherwise looks good.

Guessing the Travis errors are unrelated?

**Comment by rudigiesler** [comment]
Travis errors were unrelated. A restart fixed them.

@hodgestar I addressed your comment. Ready for review again.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**